src/NServiceBus.Core/MessageSession.cs

```diff
@@ -6,70 +6,76 @@ namespace NServiceBus
 
     class MessageSession : IMessageSession
     {
-        public MessageSession(RootContext context)
+        public MessageSession(IServiceProvider builder, MessageOperations messageOperations, PipelineCache pipelineCache, CancellationToken cancellationToken)
```

Comment:
If we follow the original convention I thinl the token should be defaulted

Reply:
still not sure I agree. This is a mandatory parameter, we expect there to be a value and we should prevent a caller from accidentally not passing one? I think the default values make sense on public APIs where users need to pass them and there are plenty of reasons they might not want to do so but in those case we expect a token and making it default sounds to undermine this assumption?

Reply:
We would have to ask the original group who added this. I think it comes from the analyzer rules and the fact that default for a cancellation token should be always considered an acceptable value because it represents a token that will never be canceled. So just from the perspective of this class the token is not required and can be considered optional.

I don't want to imply I'm not aligned with your thinking. I want to say maybe it is better to understand why it was defaulted and then make a choice whether changing or not changing is a good idea.

Reply:
Happy to further discuss this with people having more knowledge on this but I think there is no need that this conversation blocks this PR as this is internal code that can be changed easily so we can change it to something else if we find (or create) the necessary guidance.

Reply:
This was the PR https://github.com/Particular/NServiceBus/pull/5946 which confirms what I wrote above. @DavidBoike would you mind sharing your thoughts if you remember something around this?

Reply:
The analyzer forces public API to have a defaultable token. It also wants the token on private methods to be non-defaulted so the expectation of passing one is clear.

I don't know if the analyzer would object if you tried to make it defaultable in this case, given that it is not public nor private, and it's also a constructor, so it's a bit of a special case. I guess I have no opinion.

Reply:
I'm pro making the change @timbussmann suggests because it makes it explicit we always need one. Just wanted to make sure we are not missing something from the original change by moving away from a default to a mandatory parameter

Reply:
The only place you **need** a default parameter is in a public API.

Reply:
I know but as I saw it was in the other PR applied quite constantly so I wanted to double check if I missed something obvious